Dockerfile

```diff
@@ -12,14 +12,22 @@ ARG GOPROXY
 ARG KUBE_GIT_VERSION
 ENV KUBE_GIT_VERSION=$KUBE_GIT_VERSION
 
+# This will be set by buildkit automatically (e.g. to "linux").
+# Useful for building multi-arch container images.
+ARG TARGETOS
```

Comment:
If these values are constrained to be compatible with `GOOS` and `GOARCH` inputs to the go toolchain, should we just call them `GOOS` and `GOARCH`?

Reply:
Buildkit will automatically fill in the values for these build args (special names) if they are declared in your Dockerfile. When you do `docker buildx` with the multi arch flags, it will conceptually run a loop for each arch and do the equivalent of a `docker build` for each arch varying these build `ARG`s automatically.

Reply:
And when you do a regular `docker build .` (even with no additional arguments) it will still automatically fill in these specially named `ARG`s.

Reply:
Reference: https://docs.docker.com/engine/reference/builder/#automatic-platform-args-in-the-global-scope